graph.py

```diff
@@ -17,12 +39,105 @@ def __init__(self, builds):
         for build in builds:
             self.add(build)
 
+    def _get_layer_info(self, layer_id, pulp_repo_url):
+        try:
+            size = None
+            nvr = None
+            parent_id = None
+            if layer_id in self.known_pulp_layers.keys():
+                nvr = self.known_pulp_layers[layer_id]
+                size = self.found_nvr_sizes[nvr]
+            else:
+                pulp_url = pulp_repo_url + '/%s/json' % layer_id
+                r = requests.get(pulp_url, verify=False)
+                r.raise_for_status()
+                build_json = r.json()
+        except Exception:
+            return (None, None, None)
+
+        size = build_json.get('Size', 0)
+        parent_id = build_json.get('parent_id', None)
+        labels = build_json['config'].get('Labels', {'Name': '', 'Version': '', 'Release': ''})
+        nvr = '%s-%s-%s' % (labels['Name'], labels['Version'], labels['Release'])
```

Comment:
Another nitpick, but this isn't a Koji build name (that would be using the BZComponent label), and it isn't really a docker image name (that would be "%s:%s-%s"). I think it's meant to be a docker image name, so technically it should be $Name:$Version-$Release.